### PR TITLE
Enable socket and select on redox

### PR DESCRIPTION
**Pull request by coolreader18**

### CODE REVIEW

**Comment by coolreader18** [comment]
Ping; would someone be able to review this?

File: src/sys/socket/mod.rs [hunk 1]
```diff
@@ -2392,6 +2397,7 @@ pub fn shutdown(df: RawFd, how: Shutdown) -> Result<()> {
 
 #[cfg(test)]
 mod tests {
+    #[cfg(not(target_os = "redox"))]
```
Comment (asomers) [inline src/sys/socket/mod.rs:2400]: Why this line?

**Review by asomers — COMMENTED** [review]
Ok, this looks good.  Would you please add a CHANGELOG entry?

**Review by asomers — APPROVED** [review]
bors r+

**Comment by coolreader18** [comment]
Seems like a fluke?

**Comment by coolreader18** [comment]
bors retry

**Comment by coolreader18** [comment]
worth a shot :P

**Comment by asomers** [comment]
bors retry

**Pull request merged**